### DISCUSSION TITLE
Stepper servo with feedback "error"

### DISCUSSION THREAD

**Discussion by sp3swj** [gin66/FastAccelStepper#203] [Ideas]
Has anyone tried to use these stepper motors with an additional feedback signal function "fail" ? Functionally, it works just like a regular  controlled stepper motor EN/DIR/STEP but in addition, it has feedback in the form of a failure signal. If we overload the motor or lose steps, it automatically provides feedback about the error. Of course, we can use this input be handled externally in the main program, but it would be nice if this library already can serve this signal.

https://www.leadshine.com/product-detail/iCS-1708.html

I have on desktop such motor and works very nice.

( on link older version  of this stepper-servo  https://photos.app.goo.gl/62XVUTArQSiTUyfZ9 )

**Comment by gin66** [comment]
This is a very nice motor, but it is very specific. Its error signal appears only to be an alarm without further information about the number of steps lost. Anyway, I doubt direct support in FastAccelStepper is a good solution. Eventually a master library, which includes FastAccelStepper and acts on the alarm signal, may be the best approach. I could add a link to the readme for that master library.

**Reply by sp3swj** [reply]
When for example is in STOP = hold position - and you use force to move few steps  - motor try to back to oryginal position, and finaly  back :-) when you remove force. But if you move to much - motor report problem. Maybe later I demonstrate on video.

This fail signal can report three problems:
- 1 pulse  - overcurrent
- 2 pulse overvoltage
- 7 pulse position lost

can work with 2 mode selected by switch
- like classic stepper -and ignore lost step if "blocked"
- try to reach position and use more  force  - (currently about one turn)

I record video now >>   https://youtu.be/crfrhfLwzkU

we can choose what is better for our "machine"  :-)

at this moment Im very satisfied - work in progress :-)
